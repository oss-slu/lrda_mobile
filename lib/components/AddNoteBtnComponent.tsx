import React from 'react'
import { TouchableOpacity } from 'react-native'
import { View, Text, Pressable, StyleSheet } from 'react-native'
import { SvgIcon } from './SvgIcon'
import IonIcons from 'react-native-vector-icons/Ionicons'
import { useAddNoteContext } from '../context/AddNoteContext'
import { useSelector, useDispatch } from 'react-redux'
import Feather from 'react-native-vector-icons/Feather'
import { toogleAddNoteState } from '../../redux/slice/AddNoteStateSlice'
function AddNoteBtnComponent() {

    const dispatch = useDispatch();

    const {navigateToAddNote, publishNote} = useAddNoteContext();
    const appThemeColor = useSelector((state) => state?.themeSlice?.theme)
    const addNoteState = useSelector((state) => state?.addNoteState?.isAddNoteOpned);
    
    const handleAddNote = () => {
        dispatch(toogleAddNoteState())
        navigateToAddNote();

    }

    const handlePublish = () => {
        publishNote();
    }

  return (
   <View style={[styles.conatiner,]} >
        <SvgIcon style={[styles.backgroung,]}/>
        <TouchableOpacity style={styles.button} onPress={!addNoteState ? handleAddNote : handlePublish}>
            {
                !addNoteState ? 
                (<IonIcons style={styles.buttonIcon} name = 'add' size={25}/>)
                :
                (<Feather style={styles.buttonIcon} name = 'upload-cloud' size={25}/>)
            }
        </TouchableOpacity>
   </View>
  )
}


const styles = StyleSheet.create({
    conatiner: {
        position: 'relative',
        width: 75,
        alignItems: 'center'
    },
    backgroung:{
        position: 'absolute',
        backgroundColor: 'transparent'
    },
    button: {
        backgroundColor: '#f0f0f0',
        height: 50,
        width: 50,
        justifyContent: 'center',
        alignItems: 'center',
        borderRadius: 50,
        top: -25,
        shadowColor: '#000', 
        shadowOffset: { width: 0, height: 2 }, 
        shadowOpacity: 0.25, 
        shadowRadius: 4, 
        // Android elevation
        elevation: 5, 
    },
    buttonIcon: {
        fontWeight: '800',
        fontSize: 30
    },
})

export default AddNoteBtnComponent