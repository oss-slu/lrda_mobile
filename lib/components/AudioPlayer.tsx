import React, { useEffect, useState } from 'react';
import { View, Button } from 'react-native';
import { Audio } from 'expo-av';

interface AudioPlayerProps {
  audioUri: string;
}

const AudioPlayer: React.FC<AudioPlayerProps> = ({ audioUri }) => {
  const [sound, setSound] = useState<Audio.Sound | null>(null);
  const [isPlaying, setIsPlaying] = useState(false);

  useEffect(() => {
    // Load the audio when the component mounts
    const loadSound = async () => {
      const { sound } = await Audio.Sound.createAsync({ uri: audioUri });
      setSound(sound);
    };
    loadSound();

    // Cleanup the sound when the component unmounts
    return () => {
      if (sound) {
        sound.unloadAsync();
      }
    };
  }, [audioUri]);

  const handlePlayPause = async () => {
    if (sound) {
      if (isPlaying) {
        await sound.pauseAsync();
      } else {
        await sound.playAsync();
      }
      setIsPlaying(!isPlaying);
    }
  };

  return (
    <View>
      <Button title={isPlaying ? 'Pause' : 'Play'} onPress={handlePlayPause} />
    </View>
  );
};

export default AudioPlayer;
