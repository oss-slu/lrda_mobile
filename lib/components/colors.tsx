export const colors = {
  lightColors: {
    primaryColor: '#F5F9FE',
    secondaryColor: '#DCDFE4',
    tertiaryColor: '#B3B9C4',
    homeColor: '#7FADE1',
    homeBlue: '#0B1D47',
    homeGray: '#D9DDE3',
    white: '#F7F8F9',
    black: '#161A1D',
    scrollerColor: '#F7F8F9',
    text: '#161A1D',
    logout: '#22272B',
    logoutText: '#DCDFE4',
  },
  darkColors: {
    primaryColor: '#161A1D',
    secondaryColor: '#22272B',
    tertiaryColor: '#282E33',
    homeColor: '#7FADE1',
    homeBlue: '#0B1D47',
    homeGray: '#D9DDE3',
    white: '#F7F8F9',
    black: '#161A1D',
    scrollerColor: '#161A1D',
    text: '#F7F8F9',
    logout: '#DCDFE4', 
    logoutText: '#22272B',
  },
};
// #F7F8F9