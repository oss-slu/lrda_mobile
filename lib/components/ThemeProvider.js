import React, { createContext, useContext, useState, useEffect } from 'react';
import AsyncStorage from '../utils/async_storage';
import { colors } from './colors';
import { useSelector } from 'react-redux';

const ThemeContext = createContext();

export function useTheme() {
  return useContext(ThemeContext);
}

export function ThemeProvider({ children }) {

  const appThemeColor = useSelector((state) => state.themeSlice.theme);

  const [isDarkmode, setIsDarkmode] = useState(false);
  colors.darkColors.homeColor = appThemeColor;
  colors.lightColors.homeColor = appThemeColor;
  console.log(colors.darkColors.homeColor, appThemeColor)
  const toggleDarkmode = () => {
    setIsDarkmode((prevMode) => !prevMode);

    // Save the theme preference in AsyncStorage
    AsyncStorage.save('themePreference', !isDarkmode ? 'dark' : 'light');
  };

  useEffect(() => {
    // Load the user's theme preference from AsyncStorage on app start
    AsyncStorage.get('themePreference')
      .then((theme) => {
        setIsDarkmode(theme === 'dark');
      })
      .catch((error) => {
        console.error('Error loading theme preference:', error);
      });
  }, []);

  const theme = isDarkmode ? colors.darkColors : colors.lightColors;

  return (
    <ThemeContext.Provider value={{ isDarkmode, toggleDarkmode, theme }}>
      {children}
    </ThemeContext.Provider>
  );
}
