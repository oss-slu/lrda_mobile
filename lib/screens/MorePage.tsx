import React from "react";
import {
  View,
  Text,
  ScrollView,
  Linking,
  SafeAreaView,

  StyleSheet,
  Dimensions,
  Switch,
  Image,
} from "react-native";
import { useDispatch } from "react-redux";
import { useTheme } from "../components/ThemeProvider";
import { useDispatch } from "react-redux";


const user = User.getInstance();
const { width, height } = Dimensions.get("window");

export default function MorePage() {
  const { theme, isDarkmode, toggleDarkmode } = useTheme();
  const dispatch = useDispatch();

  const handleToggleDarkMode = () => {
    if (toggleDarkmode) {
      toggleDarkmode();
    }
  };

  const handleEmail = () => {
    const emailAddress = "yashkamal.bhatia@slu.edu";
    const subject = "Bug Report on 'Where's Religion?'";
    const body = "Please provide details of your issue you are facing here.";
    const emailUrl = `mailto:${emailAddress}?subject=${encodeURIComponent(subject)}&body=${encodeURIComponent(body)}`;
    Linking.openURL(emailUrl);
  };

  const handleVisitWebsite = () => {
    const websiteUrl = "https://www.wheresreligion.org";
    Linking.openURL(websiteUrl);
  };

  const onLogoutPress = async () => {
    try {
      await user.logout(dispatch);
    } catch (e) {
      console.log(e);
    }
  };

  return (
    <>
      <View style={[styles.header, { backgroundColor: theme.background }]}>
        <Text style={[styles.headText, { color: theme.text }]}>About</Text>
      </View>
      <SafeAreaView>
        <ScrollView contentContainerStyle={[styles.container, { paddingBottom: 140 }]}>
          {/* About Section Content */}
          <View style={styles.textContainer}>
            <Text style={[styles.titleText,{color:theme.text}]}>What is Where’s Religion?</Text>
            <Text style={[styles.bodyText, { color: theme.text }]}>
            Where’s Religion? is an open-source application developed by humanities faculty and IT professionals at Saint Louis University that supports in-person research, remote data entry, media sharing, and mapping. The app is designed to facilitate a more robust public understanding of religion through rigorous scholarly methods. Our conviction is that the study of religion must account for the wide range of embodied experiences, improvised practices, material cultures, and shared spaces that humans inhabit. Through a research methodology that moves beyond analysis of sacred texts, creeds, and official teachings, Where’s Religion? provides a platform to diversify the data we study and to advance the study of religion we all encounter in everyday life.</Text>
            <Text style={[styles.bodyText, { color: theme.text }]}>
Where’s Religion? is a keystone outcome of the Center on Lived Religion at Saint Louis University. We have received external support from the Henry Luce Foundation ($400,000 in 2018 and $470,000 in 2022), and internal support from the College of Arts & Sciences, the Office for the Vice President for Research and the Research Computing Group, Open Source with SLU, the Walter J. Ong, S.J., Center for Digital Humanities, and the CREST Research Center (Culture, Religion, Ethics, Science, Technology).</Text>
          </View>

          {/* Initiative Team Section */}
          <View style={styles.initiativeTeamSection}>
            
            <View style={styles.initiativeTeamMembers}>
            <View style={styles.teamMemberContainer}>
  {Image.resolveAssetSource(require('../../assets/Rachel.jpg')).uri ? (
    <Image source={require('../../assets/Rachel.jpg')} style={styles.teamImage} />
  ) : (
    <Text>Image not available</Text>
  )}
  <Text style={[styles.teamName,{color:theme.text}]}>Rachel Lindsey</Text>
  <Text style={[styles.teamRole,{color:theme.text}]}>Director of Center on Lived Religion</Text>
</View>

              <View style={styles.initiativeMemberContainer}>
              {Image.resolveAssetSource(require('../../assets/Adam.jpg')).uri ? (
    <Image source={require('../../assets/Adam.jpg')} style={styles.teamImage} />
  ) : (
    <Text>Image not available</Text>
  )}
                <Text style={[styles.initiativeMemberName,{color:theme.text}]}>Adam Park</Text>
                <Text style={[styles.initiativeMemberRole,{color:theme.text}]}>Associate Director of Research (COLR)</Text>
              </View>
            </View>
            <View style={styles.initiativeTeamMembers}>
              <View style={styles.teamMemberContainer}>
              {Image.resolveAssetSource(require('../../assets/Yash.jpg')).uri ? (
    <Image source={require('../../assets/Yash.jpg')} style={styles.teamImage} />
  ) : (
    <Text>Image not available</Text>
  )}
                <Text style={[styles.teamName,{color:theme.text}]}>Yash Bhatia</Text>
                <Text style={[styles.teamRole,{color:theme.text}]}>Software Engineer and Tech Lead </Text>
              </View>
            </View>
          </View>

          {/* Development Team Section */}
          <View style={styles.teamSection}>
            <Text style={styles.teamTitle}>The Development Team</Text>
            <View style={styles.teamMembersRow}>
              <View style={styles.teamMemberContainer}>
              {Image.resolveAssetSource(require('../../assets/Patrick.jpg')).uri ? (
    <Image source={require('../../assets/Patrick.jpg')} style={styles.teamImage} />
  ) : (
    <Text>Image not available</Text>
  )}
                <Text style={[styles.teamName,{color:theme.text}]}>Patrick Cuba</Text>
                <Text style={[styles.teamRole,{color:theme.text}]}>IT Architect</Text>
              </View>
              <View style={styles.teamMemberContainer}>
              {Image.resolveAssetSource(require('../../assets/Bryan.jpg')).uri ? (
    <Image source={require('../../assets/Bryan.jpg')} style={styles.teamImage} />
  ) : (
    <Text>Image not available</Text>
  )}
                <Text style={[styles.teamName,{color:theme.text}]}>Bryan Haberberger</Text>
                <Text style={[styles.teamRole,{color:theme.text}]}>Full Stack Developer</Text>
              </View>
            </View>
            <View style={styles.teamMembersRow}>
              <View style={styles.teamMemberContainer}>
              {Image.resolveAssetSource(require('../../assets/Stuart.jpg')).uri ? (
    <Image source={require('../../assets/Stuart.jpg')} style={styles.teamImage} />
  ) : (
    <Text>Image not available</Text>
  )}
                <Text style={[styles.teamName,{color:theme.text}]}>Stuart Ray</Text>
                <Text style={[styles.teamRole,{color:theme.text}]}>Developer</Text>
              </View>
              <View style={styles.teamMemberContainer}>
              {Image.resolveAssetSource(require('../../assets/Izak.jpg')).uri ? (
    <Image source={require('../../assets/Izak.jpg')} style={styles.teamImage} />
  ) : (
    <Text>Image not available</Text>
  )}
                <Text style={[styles.teamName,{color:theme.text}]}>Izak Robles</Text>
                <Text style={[styles.teamRole,{color:theme.text}]}>Developer</Text>
              </View>
            </View>
          </View>

          {/* Dark Mode and Logout at the End of ScrollView */}
          <View style={styles.bottomContainer}>
          <View
              style={[
                styles.switchContainer,
                { backgroundColor: theme.background },
              ]}
            >
              <Text style={[styles.switchText, { color: theme.text }]}testID="Light Mode ">
                {isDarkmode ? "Light Mode" : "Dark Mode"}
              </Text>
              <Switch
                testID="dark-mode-switch"
                trackColor={{
                  false: "black",
                  true: theme.text,
                }}
                thumbColor={theme.primaryColor}
                onValueChange={handleToggleDarkMode}
                value={isDarkmode}
              />
            </View>

            <TouchableOpacity
              style={[
                styles.button,
                { backgroundColor: theme.primaryColor },
              ]}
              onPress={handleEmail}
            >
              <Text
                style={[
                  styles.buttonText,
                  { color: theme.text },
                ]}
              >
                Report
              </Text>
              <Ionicons
                name="mail-outline"
                size={24}
                color={theme.text}
                style={{ marginLeft: 10 }}
              />
            </TouchableOpacity>

            <TouchableOpacity
              style={[
                styles.button,
                { backgroundColor: theme.primaryColor },
              ]}
              onPress={handleVisitWebsite}
            >
              <Text
                style={[
                  styles.buttonText,
                  { color: theme.text },
                ]}
              >
                Visit Our Website
              </Text>
              <Ionicons
                name="globe-outline"
                size={24}
                color={theme.text}
                style={{ marginLeft: 10 }}
              />
            </TouchableOpacity>

            <TouchableOpacity
              key="Logout"
              style={[
                styles.logout,
                { backgroundColor: isDarkmode ? "white" : "black" },
              ]}
              onPress={onLogoutPress}
            >
              <Text
                style={[
                  styles.logoutText,
                  { color: isDarkmode ? "black" : "white" },
                ]}
              >
                Logout
              </Text>
              <Ionicons
                name={"log-out-outline"}
                size={30}
                color={isDarkmode ? "black" : "white"}
              />
            </TouchableOpacity>
          </View>
        </ScrollView>
      </SafeAreaView>
    </>
  );
}

const styles = StyleSheet.create({
  header: { paddingVertical: height * 0.02, justifyContent: "center", alignItems: "center" },
  headText: { fontSize: 32, fontWeight: "bold", paddingTop: 40 },
  container: { flexGrow: 1, alignItems: "center", paddingHorizontal: 20 },
  textContainer: { width: "100%", alignItems: "center", marginTop: 20 },
  titleText: { fontSize: 28, fontWeight: "600", textAlign: "center", marginBottom: 20 },
  bodyText: { fontSize: 16, lineHeight: 24, textAlign: "justify", marginBottom: 15, paddingHorizontal: 10 },
  button: {
    flexDirection: "row",
    alignItems: "center",
    justifyContent: "center",
    height: 50,
    width: "90%",
    borderRadius: 15,
    marginTop: 10,
    paddingHorizontal: 20,
  },
  buttonText: {
    fontSize: 20,
    fontWeight: "600",
    textAlign: "center",
  },


  // Team Section
  teamSection: { width: "100%", padding: 20, alignItems: "center" },
  teamTitle: { fontSize: 24, fontWeight: "bold", marginBottom: 16 },
  teamMembersRow: {
    flexDirection: 'row',
    justifyContent: 'space-around',
    width: '100%',
    marginBottom: 16,
  },
  teamMemberContainer: { alignItems: 'center', width: '45%' },
  teamImage: { width: 80, height: 80, borderRadius: 40, marginBottom: 8 },
  teamName: { fontSize: 16, fontWeight: "bold", textAlign: "center" },
  teamRole: { fontSize: 14, textAlign: "center" },

  // Initiative Team Section
  initiativeTeamSection: { width: "100%", padding: 20, alignItems: "center" },
  initiativeTeamTitle: { fontSize: 24, fontWeight: "bold", marginBottom: 16 },
  initiativeTeamMembers: { flexDirection: "row", justifyContent: "space-around", width: "100%", marginBottom: 16 },
  initiativeMemberContainer: { alignItems: "center", width: "45%" },
  initiativeMemberName: { fontSize: 16, fontWeight: "bold", textAlign: "center" },
  initiativeMemberRole: { fontSize: 14, textAlign: "center" },

  // Bottom Container for Dark Mode and Logout
  bottomContainer: {
    width: '100%',
    alignItems: 'center',
    marginTop: 20,
  },
  switchContainer: {
    flexDirection: "row",
    justifyContent: "space-between",
    alignItems: "center",
    width: "90%",
    padding: 10,
    borderRadius: 10,
    marginBottom: 10,
  },
  switchText: { fontSize: 18, fontWeight: "500" },
  logout: { flexDirection: "row", justifyContent: "center", alignItems: "center", height: 50, width: "90%", borderRadius: 15, marginTop:10},
  logoutText: { fontSize: 20, fontWeight: "600", marginRight: 10 },
});
