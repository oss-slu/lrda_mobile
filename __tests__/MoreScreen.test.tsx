import { fireEvent, render } from '@testing-library/react-native';
import moxios from 'moxios';
import React from 'react';
import { Linking } from 'react-native';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { useTheme } from '../lib/components/ThemeProvider';
import MorePage from '../lib/screens/MorePage';

// Create a mock Redux store
const mockStore = configureStore([]);
const store = mockStore({
  navigation: {
    navState: 'more', // Add mock navigation state if needed
  },
  theme: {
    darkMode: false, // Add mock theme state if needed
  },
});

// Mock the ThemeProvider with spies
jest.mock('../lib/components/ThemeProvider', () => ({
  useTheme: jest.fn(() => ({
    theme: {
      primaryColor: '#ffffff',
      text: '#000000',
      secondaryColor: '#f0f0f0',
      logout: '#ff0000',
      logoutText: '#ffffff',
    },
    isDarkmode: false, // Initial value of isDarkmode
    toggleDarkmode: jest.fn(), // Mock the toggleDarkmode function
  })),
}));

// Mock the User class
jest.mock('../lib/models/user_class', () => {
  return {
    User: {
      getInstance: jest.fn(() => ({
        logout: jest.fn(),
      })),
    },
  };
});

beforeAll(() => {
  // Suppress console logs during tests
  jest.spyOn(console, 'log').mockImplementation(() => {});
  jest.spyOn(console, 'error').mockImplementation(() => {});

  // Install moxios for mocking HTTP requests
  moxios.install();
});

afterAll(() => {
  // Restore the original console methods and uninstall moxios
  console.log.mockRestore();
  console.error.mockRestore();
  moxios.uninstall();
});

describe('MorePage', () => {
  it('renders correctly', () => {
    const { toJSON } = render(
      <Provider store={store}>
        <MorePage />
      </Provider>
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it('toggles dark mode correctly', () => {
    const { getByTestId } = render(
      <Provider store={store}>
        <MorePage />
      </Provider>
    );

    // Get the Switch component with testID "dark-mode-switch"
    const toggleSwitch = getByTestId('dark-mode-switch');

    // Simulate the toggle of dark mode
    fireEvent(toggleSwitch, 'onValueChange', true);

    // Ensure that dark mode is toggled
    expect(toggleSwitch.props.value).toBe(false); // Based on initial value of `isDarkmode` being false
  });

  it("opens email link when 'Report a Bug' is pressed", () => {
    const spy = jest.spyOn(Linking, 'openURL');

    const { getByText } = render(
      <Provider store={store}>
        <MorePage />
      </Provider>
    );

    // Find the 'Report a Bug' button and simulate press
    const emailButton = getByText('Report a Bug');
    fireEvent.press(emailButton);

    // Ensure the correct email URL is opened
    expect(spy).toHaveBeenCalledWith(
      "mailto:yashkamal.bhatia@slu.edu?subject=Bug%20Report%20on%20'Where's%20Religion%3F'&body=Please%20provide%20details%20of%20your%20issue%20you%20are%20facing%20here."
    );
  });

  it('displays correct theme text based on isDarkmode value', () => {
    const { getByText, rerender } = render(
      <Provider store={store}>
        <MorePage />
      </Provider>
    );
  
    // Check Light Mode text when isDarkmode is false
    expect(getByText('Light Mode')).toBeTruthy();
  
    // Update isDarkmode to true & renderer
    useTheme.mockReturnValueOnce({
      theme: {
        primaryColor: '#ffffff',
        text: '#ffffff',
        secondaryColor: '#f0f0f0',
        logout: '#ff0000',
        logoutText: '#ffffff',
      },
      isDarkmode: true,
      toggleDarkmode: jest.fn(),
    });
    rerender(
      <Provider store={store}>
        <MorePage />
      </Provider>
    );
  
    expect(getByText('Dark Mode')).toBeTruthy();
  });

  it('displays correct text color for theme text', () => {
    const { getByText, rerender } = render(
      <Provider store={store}>
        <MorePage />
      </Provider>
    );
  
    // Check that Light Mode text is black when isDarkmode is false
    const lightModeText = getByText('Light Mode');
    expect(lightModeText.props.style).toEqual(expect.arrayContaining([{ color: 'black' }]));
  
    // Update isDarkmode to true and rerender
    useTheme.mockReturnValueOnce({
      theme: {
        primaryColor: '#ffffff',
        text: '#ffffff',
        secondaryColor: '#f0f0f0',
        logout: '#ff0000',
        logoutText: '#ffffff',
      },
      isDarkmode: true,
      toggleDarkmode: jest.fn(),
    });
    rerender(
      <Provider store={store}>
        <MorePage />
      </Provider>
    );
  
    // Check that Dark Mode text is white when isDarkmode is true
    const darkModeText = getByText('Dark Mode');
    expect(darkModeText.props.style).toEqual(expect.arrayContaining([{ color: 'white' }]));
  });
});
