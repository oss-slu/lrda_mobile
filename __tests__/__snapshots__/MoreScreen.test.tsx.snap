// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MorePage renders correctly 1`] = `
[
  <View
    style={
      [
        {
          "alignItems": "center",
          "backgroundColor": "#ffffff",
          "justifyContent": "center",
          "marginBottom": -8,
          "marginTop": 53.36,
          "padding": 13.34,
        },
        {
          "backgroundColor": undefined,
        },
      ]
    }
  >
    <Text
      style={
        [
          {
            "color": "#000000",
            "fontSize": 32,
            "fontWeight": "bold",
          },
          {
            "color": "#000000",
          },
        ]
      }
    >
      Where's Religion
    </Text>
  </View>,
  <RCTSafeAreaView>
    <RCTScrollView
      contentContainerStyle={
        {
          "alignItems": "center",
          "flexGrow": 1,
          "justifyContent": "flex-start",
        }
      }
    >
      <View>
        <View
          style={
            {
              "height": 300,
              "width": "100%",
            }
          }
        >
          <Image
            source={1}
            style={
              {
                "flex": 1,
                "height": undefined,
                "resizeMode": "cover",
                "width": undefined,
              }
            }
          />
        </View>
        <View>
          <Text
            style={
              [
                {
                  "alignSelf": "center",
                  "color": "#000000",
                  "fontSize": 28,
                  "fontWeight": "bold",
                },
                {
                  "color": "#000000",
                },
              ]
            }
          >
            Resources
          </Text>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "opacity": 1,
              }
            }
          >
            <Text
              style={
                [
                  {
                    "alignSelf": "center",
                    "color": "#000000",
                    "fontSize": 16,
                    "lineHeight": 28,
                  },
                  {
                    "color": "#000000",
                  },
                ]
              }
            >
              Our Website
            </Text>
          </View>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "opacity": 1,
              }
            }
          >
            <Text
              style={
                [
                  {
                    "alignSelf": "center",
                    "color": "#000000",
                    "fontSize": 16,
                    "lineHeight": 28,
                  },
                  {
                    "color": "#000000",
                  },
                ]
              }
            >
              Guide to Ethnography
            </Text>
          </View>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "opacity": 1,
              }
            }
          >
            <Text
              style={
                [
                  {
                    "alignSelf": "center",
                    "color": "#000000",
                    "fontSize": 16,
                    "lineHeight": 28,
                  },
                  {
                    "color": "#000000",
                  },
                ]
              }
            >
              Guide to Coding
            </Text>
          </View>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "opacity": 1,
              }
            }
          >
            <Text
              style={
                [
                  {
                    "alignSelf": "center",
                    "color": "#000000",
                    "fontSize": 16,
                    "lineHeight": 28,
                  },
                  {
                    "color": "#000000",
                  },
                ]
              }
            >
              Report a Bug
            </Text>
          </View>
          <Text
            style={
              [
                {
                  "alignSelf": "center",
                  "color": "#000000",
                  "fontSize": 28,
                  "fontWeight": "bold",
                },
                {
                  "color": "#000000",
                },
              ]
            }
          >
            Meet our Team
          </Text>
          <Text
            style={
              [
                {
                  "alignSelf": "center",
                  "color": "#000000",
                  "fontSize": 16,
                  "lineHeight": 28,
                },
                {
                  "color": "#000000",
                },
              ]
            }
          >
            Insert Team Photo
          </Text>
          <Text
            style={
              [
                {
                  "alignSelf": "center",
                  "color": "#000000",
                  "fontSize": 16,
                  "lineHeight": 28,
                },
                {
                  "color": "#000000",
                },
              ]
            }
          >
            Insert Team Message
          </Text>
          <Text
            style={
              [
                {
                  "alignSelf": "center",
                  "color": "#000000",
                  "fontSize": 28,
                  "fontWeight": "bold",
                },
                {
                  "color": "#000000",
                },
              ]
            }
          >
            Frequently Asked Questions
          </Text>
        </View>
        <Text
          style={
            [
              {
                "alignSelf": "center",
                "color": "#000000",
                "fontSize": 28,
                "fontWeight": "bold",
              },
              {
                "color": "#000000",
              },
            ]
          }
        >
          What can users do?
        </Text>
        <Text
          style={
            [
              {
                "alignSelf": "center",
                "color": "#000000",
                "fontSize": 16,
                "lineHeight": 28,
              },
              {
                "color": "#000000",
              },
            ]
          }
        >
          Explore religious traditions, find places of worship, engage in meaningful discussions.
        </Text>
        <Text
          style={
            [
              {
                "alignSelf": "center",
                "color": "#000000",
                "fontSize": 28,
                "fontWeight": "bold",
              },
              {
                "color": "#000000",
              },
            ]
          }
        >
          Who is it for?
        </Text>
        <Text
          style={
            [
              {
                "alignSelf": "center",
                "color": "#000000",
                "fontSize": 16,
                "lineHeight": 28,
              },
              {
                "color": "#000000",
              },
            ]
          }
        >
          Scholars, students, believers, and the curious about the world's religions.
        </Text>
        <Text
          style={
            [
              {
                "alignSelf": "center",
                "color": "#000000",
                "fontSize": 28,
                "fontWeight": "bold",
              },
              {
                "color": "#000000",
              },
            ]
          }
        >
          What's unique?
        </Text>
        <Text
          style={
            [
              {
                "alignSelf": "center",
                "color": "#000000",
                "fontSize": 16,
                "lineHeight": 28,
              },
              {
                "color": "#000000",
              },
            ]
          }
        >
          Provides a modern method to capture experiences using the devices that are with us every day.
        </Text>
        <Text
          style={
            [
              {
                "alignSelf": "center",
                "color": "#000000",
                "fontSize": 28,
                "fontWeight": "bold",
              },
              {
                "color": "#000000",
              },
            ]
          }
        >
          Our Mission
        </Text>
        <Text
          style={
            [
              {
                "alignSelf": "center",
                "color": "#000000",
                "fontSize": 16,
                "lineHeight": 28,
              },
              {
                "color": "#000000",
              },
            ]
          }
        >
          Connect people of diverse religious backgrounds, beliefs, and practices.
        </Text>
        <Text
          style={
            [
              {
                "alignSelf": "center",
                "color": "#000000",
                "fontSize": 28,
                "fontWeight": "bold",
              },
              {
                "color": "#000000",
              },
            ]
          }
        >
          Why use 'Where's Religion?'
        </Text>
        <Text
          style={
            [
              {
                "alignSelf": "center",
                "color": "#000000",
                "fontSize": 16,
                "lineHeight": 28,
              },
              {
                "color": "#000000",
              },
            ]
          }
        >
          Explore religious traditions, find places of worship, engage in meaningful discussions.
        </Text>
        <View
          style={
            {
              "paddingTop": 15,
              "width": "100%",
            }
          }
        >
          <View
            style={
              {
                "alignItems": "center",
                "marginTop": 40,
              }
            }
          >
            <View
              style={
                [
                  {
                    "alignItems": "center",
                    "backgroundColor": "#f0f0f0",
                    "borderRadius": 10,
                    "flexDirection": "row",
                    "justifyContent": "space-between",
                    "padding": 6,
                    "width": "94%",
                  },
                  {
                    "backgroundColor": undefined,
                  },
                ]
              }
            >
              <Text
                style={
                  [
                    {
                      "color": "#000000",
                      "fontSize": 18,
                      "fontWeight": "500",
                      "marginLeft": 9,
                    },
                    {
                      "color": "#000000",
                    },
                  ]
                }
              >
                Light Mode
              </Text>
              <RCTSwitch
                accessibilityRole="switch"
                onChange={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                onTintColor="#000000"
                style={
                  {
                    "height": 31,
                    "width": 51,
                  }
                }
                testID="dark-mode-switch"
                thumbTintColor="#ffffff"
                tintColor="black"
                value={false}
              />
            </View>
          </View>
        </View>
        <View
          style={
            [
              {
                "alignItems": "center",
                "backgroundColor": "#ffffff",
                "width": "100%",
              },
              {
                "backgroundColor": undefined,
              },
            ]
          }
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "black",
                "borderRadius": 15,
                "flexDirection": "row",
                "height": 50,
                "justifyContent": "center",
                "marginBottom": 100,
                "marginTop": 60,
                "opacity": 1,
                "width": "80%",
              }
            }
          >
            <Text
              style={
                [
                  {
                    "color": "#ffffff",
                    "fontSize": 20,
                    "fontWeight": "600",
                    "marginLeft": 5,
                    "marginRight": 10,
                  },
                  {
                    "color": "white",
                  },
                ]
              }
            >
              Logout
            </Text>
            <Text />
          </View>
        </View>
      </View>
    </RCTScrollView>
  </RCTSafeAreaView>,
]
`;
